main.cpp

```diff
@@ -213,6 +412,34 @@ struct CarWash //                                   1) define an empty struct fo
     3) controll instruments
  */
 
+
+struct Midi       
+{
+    //has value
+    int value = 127; //3) member variables with relevant data types.
+    //has channel
+    int channel = 127; //3) member variables with relevant data types.
+    //has input
+    int input = 127; //3) member variables with relevant data types.
+    //has output
+    int output = 127; //3) member variables with relevant data types.
+    //has sync type
+    int syncType = 127; //3) member variables with relevant data types.
+
+    /* 
+    member functions with a user-defined type as the parameter.
+    The user-defined type parameter happens to be the nested class.
+    */
+
+    //2) send midi
+    void sendMidi();
+    //2) receive midi
+    void receiveMidi();
+    //2) controll instruments
+    void controllInstrument();
```

Comment:
`Midi::sendMidi()`...   What you want is `MidiInterface::sendMidi(MidiMessage)`
that means 2 classes (here's a chance for your nested class) are needed here.
- [ ] revise